internal/users/db/db_test.go

```diff
@@ -131,8 +131,8 @@ func TestMigrationToLowercaseUserAndGroupNames(t *testing.T) {
 	t.Cleanup(func() { db.SetGroupFile(origGroupFile) })
 
 	// Make the userutils package to use test locking for the group file
-	userutils.OverrideShadowPasswordLocking()
-	t.Cleanup(userutils.RestoreShadowPasswordLocking)
+	localentries.OverrideLocking()
+	t.Cleanup(localentries.RestoreLocking)
```

Comment:
> userutils: Move local database locking into localentries
> 
> localentries contains types for handling local users and groups and
> that's where locking is needed, so use move the locking user utilities
> here instead.

Since we plan to use the locking when we pick a UID/GID for an authd user/group (UDENG-6745), I think the locking fits better into the more generic `userutils` package than the `localentries` package. If you prefer to have it below `internal/users/`, I would also be fine with putting into a separate `internal/users/locking/` package.

Reply:
> Since we plan to use the locking when we pick a UID/GID for an authd user/group (UDENG-6745), I think the locking fits better into the more generic userutils package than the localentries package. If you prefer to have it below internal/users/, I would also be fine with putting into a separate internal/users/locking/ package.

I liked the fact to use the package as the clear statement of what this locks (not only users though), so given it locks the local entries (and potentially other NSS modules acting on them), I thought it was the right spot.

I'm fine to move it to somewhere else too, but maybe `locking` is too generic too, so maybe `internal/users/pwdlocking` or (even not inside `users`)?

Reply:
Ok, moved to `users/locking` but preserved the namespace as `userslocking` so that it's not confused when used outside users namespace (as per @didrocks suggestion too).